modules/renter/hostdb/scan.go

```diff
@@ -46,14 +46,54 @@ var (
 // The gofunc is created inside of this function to eliminate the burden of
 // needing to remember to call 'go addHostToScanPool'.
 func (hdb *HostDB) scanHostEntry(entry *hostEntry) {
+	// Add the entry to a waitlist, then check if any thread is currently
+	// emptying the waitlist. If not, spawn a thread to empty the waitlist.
```

Comment:
This was (I felt) the quickest way to handle this. The hostdb is one of the earlier modules, so it doesn't have much of the safety and it really doesn't have any of the clean shutdown.

Doing this properly will take a few days at least, and I want to be focusing my energy on the renter at this point, because for the most part the hostdb is performing just fine.